apis/deployer/utils/continuousreconcile/types.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package continuousreconcile
+
+import (
+	"strings"
+
+	lscore "github.com/gardener/landscaper/apis/core"
+)
+
+// ContinuousReconcileSpec represents the specification of a continuous reconcile schedule.
+type ContinuousReconcileSpec struct {
+	// Cron is a standard crontab specification of the reconciliation schedule.
+	// Either Cron or Every has to be specified.
+	// +optional
+	Cron string `json:"cron,omitempty"`
+
+	// Every specifies a delay after which the reconcile should happen.
+	// Either Cron or Every has to be specified.
+	// +optional
+	Every *lscore.Duration `json:"every,omitempty"`
+}
+
+func (crs ContinuousReconcileSpec) String() string {
+	var sb strings.Builder
+	sb.WriteString("{")
+	if len(crs.Cron) != 0 {
+		sb.WriteString("Cron: \"")
+		sb.WriteString(crs.Cron)
+		sb.WriteString("\"")
+		if crs.Every != nil && crs.Every.Duration != 0 {
+			sb.WriteString(", ")
+		}
+	}
+	if crs.Every != nil && crs.Every.Duration != 0 {
```

Comment:
The stringer method will be called by the validation for creating the error message, so it could be the case that both fields are set (in which case the spec would be invalid).
The reason for the 'complexity' is the pointer topic explained above.